Lib/fontTools/ttLib/tables/_g_l_y_f.py

```diff
@@ -139,9 +142,8 @@ def toXML(self, writer, ttFont, progress=None, splitGlyphs=False):
 			glyph = self[glyphName]
 			if glyph.numberOfContours:
 				if splitGlyphs:
-					path, ext = os.path.splitext(writer.file.name)
-					fileNameTemplate = path + ".%s" + ext
-					glyphPath = fileNameTemplate % nameToIdentifier(glyphName)
+					glyphPath = userNameToFileName(unicode(glyphName, 'utf-8'), prefix=("%s%s"%(path, "_")), suffix=ext)
```

Comment:
It is nice to have an extra '_' prior to the glyph name so that the table .ttx appears in the directory listing as the first file (i.e. font_name._g_l_y_f.ttx) and not mixed with all of the other _g_l_y_f_ files...

Reply:
I am not sure about that `_` separating the `_g_l_y_f` table tag from the glyph name. It's not clear where the table tag ends and where the glyph name begins. A dot could be better in that regard (though there's `.notdef` or `.null` which may look odd with two dots).

You know, I'm thinking that perhaps for the second split level (the first being the individual tables) we group the files in a sub-directory instead of appending a suffix (the glyph name) to the table file name.

So when -g option (splitGlyphs) is used, the glyf table is dumped like that:

- the individual glyphs' ttx files are dumped inside a new sudirectory called FontName._g_l_y_f (or FontName#1._g_l_y_f etc. when dumped more than once to the same destination)
- an index file named FontName._g_l_y_f.ttx contains the list of filenames; it doesn't need to verbosely specify the sub-path, but it can just presume that the glyph files are under a subdirectory called FontName._g_l_y_f.
i.e. instead of the current:

```xml
  <glyf>
    <TTGlyph src="CALIBRI._g_l_y_f_.notdef.ttx"/>
    <TTGlyph src="CALIBRI._g_l_y_f_A_.ttx"/>
```

we could have something like:

```xml
  <glyf>
    <TTGlyph src=".notdef.ttx"/>
    <TTGlyph src="A_.ttx"/>
```

The index file FontName._g_l_y_f.ttx and the subdirectory FontName._g_l_y_f will nicely show up next to each other in the directory's file listing, and the thousands of glyph files will not be mixed up with the rest of the individual tables, but will be logically grouped together.

WDYT?

Reply:
I think that using a directory name for the prefix is a great idea.  I think that we might want to leave the directory name in the index table as well though so that it is explicitly readable without somewhat hidden context information.

i.e.:

`  <glyf>
    <TTGlyph src="Fontname._g_l_y_f/.notdef.ttx"/>
    <TTGlyph src="Fontname._g_l_y_f/A_.ttx"/>
`
This may also keep the code (xmlReader) a bit simpler - again not needing to pull context information - always treating 'src' values the same.